pyomo/repn/plugins/nl_writer.py

```diff
@@ -1315,7 +1339,7 @@ def write(self, model):
         # "F" lines (external function definitions)
         #
         amplfunc_libraries = set()
-        for fid, fcn in sorted(self.external_functions.values()):
+        for fid, fcn in self.external_functions:
```

Comment:
Does sorting no longer matter here?

Reply:
The dict is created in the sorted order, and as of Python 3.7, dicts are ordered, so sorting was unnecessary.

Reply:
Ah yes, of course, thanks, Python.